src/infill.cpp

```diff
@@ -844,55 +882,48 @@ void Infill::connectLines(Polygons& result_lines)
                     assert(crossing_handle != (size_t)-1);
                     const size_t previous_crossing_handle = connected_lines.find(previous_crossing);
                     assert(previous_crossing_handle != (size_t)-1);
-                    if (crossing_handle
-                        == previous_crossing_handle) // These two infill lines are already connected. Don't create a loop now. Continue connecting with the next crossing.
+                    if (crossing_handle == previous_crossing_handle)
                     {
+                        // These two infill lines are already connected. Don't create a loop now. Continue connecting with the next crossing.
                         continue;
                     }
 
                     // Join two infill lines together with a connecting line.
                     // Here the InfillLineSegments function as a linked list, so that they can easily be joined.
-                    const Point previous_point
-                        = (previous_segment->start_segment == vertex_index && previous_segment->start_polygon == polygon_index) ? previous_segment->start : previous_segment->end;
-                    const Point next_point = (crossing->start_segment == vertex_index && crossing->start_polygon == polygon_index) ? crossing->start : crossing->end;
+                    const bool previous_forward = (previous_segment->start_segment == vertex_index && previous_segment->start_polygon == polygon_index);
+                    const bool next_forward = (crossing->start_segment == vertex_index && crossing->start_polygon == polygon_index);
+                    Point& previous_point = previous_forward ? previous_segment->start : previous_segment->end;
+                    Point& next_point = next_forward ? crossing->start : crossing->end;
+
                     InfillLineSegment* new_segment;
-                    // If the segment is zero length, we avoid creating it but still want to connect the crossing with the previous segment
+                    // If the segment is near zero length, we avoid creating it but still want to connect the crossing with the previous segment.
```

Comment:
```suggestion
                    // If the segment is near length, we avoid creating it but still want to connect the crossing with the previous segment.
```
Since the resolve this comment doesn't describe the behavior anymore.